### ISSUE TITLE
Lowering the attack surface of dom0-update on sys-firewall

### ISSUE THREAD

**Issue by ydirson** [QubesOS/qubes-issues#6984]
dnf processing is quite heavyweight (and that probably comes with some attack surface), and launching it in sys-firewall can probably be avoided.

The offline-download job that sys-firewall executes on dom0's request is indeed not a novel idea, and was implemented in the past using common download tools on the internet-connected side, eg. for Debian derivatives in apt-zip, whose memory [apparently lives on for convenience](http://manpages.ubuntu.com/manpages/precise/man8/apt-zip-inst.8.html).

**Comment by DemiMarie** [comment]
> > If you don't mind more memory usage, you can use a different qube for "updatevm", for example a disposable one, similar to https://www.qubes-os.org/doc/disposable-customization/#create-custom-sys-net-sys-firewall-and-sys-usb-disposables
> 
> +1. This is exactly what I use with an auto shutdown script to lower the memory usage.

I do that as well, and use a DispVM for it with no network access (it uses the qrexec proxy via sys-whonix).  I would be willing to contribute this upstream.

**Comment by ben-grande** [comment]
> > If you don't mind more memory usage, you can use a different qube for "updatevm", for example a disposable one, similar to https://www.qubes-os.org/doc/disposable-customization/#create-custom-sys-net-sys-firewall-and-sys-usb-disposables
> 
> Should this issue be about making this the default?

I don't think it can use a disposable template as an updatevm to generate a disposable of it. Might be a solution, otherwise, would be to use named disposable. I think the reason more qubes are not included in Qubes is memory constraints to deal with restricted environments.

@marmarek Is a new `sys-update` formula/playbook ideal to be called on the installer if there is more than 32GB of RAM?

**Comment by marmarek** [comment]
Opt-in sys-update is a good idea. Would be also useful when sys-firewall is not Linux at all (like, qubes-migrage-firewall).
